### ISSUE TITLE
Always getting Nan output readings when using the TCTemp function for thermocouple setup

### ISSUE THREAD

**Issue by caldecy** [arduino-libraries/Arduino_PortentaMachineControl#12]
Hi, I tried compiling the sample code for Thermocouple and upon checking the serial monitor output, it only reads "nan". I've checked that the thermocouple is functional, connected properly to CH0 and supplied with adequate power. I'm using the Portenta AKX00032 device to read the thermocouple values. Any help would be appreciated! Thank you.

**Comment by leonardocavagnis** [comment]
Hi @caldecy,
Please, could you provide additional details?
For instance, a photo of physical connection, temperature probe model and everything that could be useful to better understand the issue
Thanks

**Comment by caldecy** [comment]
Hi @leonardocavagnis, 
Apologies for the late reply on the issue. 
The photo of my thermocouple connection to the board is as shown: 
[Connection.pdf](https://github.com/arduino-libraries/Arduino_PortentaMachineControl/files/15271862/Connection.pdf)

Thermocouple model: 
[Thermocouple Model.pdf](https://github.com/arduino-libraries/Arduino_PortentaMachineControl/files/15271868/Thermocouple.Model.pdf)

The code that I use is the [Temp_probes_Thermocouples.ino](https://github.com/arduino-libraries/Arduino_PortentaMachineControl/blob/main/examples/Temp_probes_Thermocouples/Temp_probes_Thermocouples.ino)

And the result I get is "_nan_" for all channels' output.

**Comment by leonardocavagnis** [comment]
Thanks for reply!
What library version are you using?
Please, verify you're using the latest (1.0.1).

**Comment by caldecy** [comment]
Yes I am using 1.0.1 when I first encountered the issue. I tried using 1.0.0 too but faced the same issue.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by leonardocavagnis** [comment]
Did you also try the old version of [Portenta Machine Control library](https://github.com/arduino-libraries/Arduino_MachineControl)?
(not suggested to use the deprecated one, but just for double checking)

**Comment by caldecy** [comment]
Yes, and that’s the only way that worked for my setup and project.
________________________________

**Comment by leonardocavagnis** [comment]
Ok, thanks!
what's the version of the core?
![image](https://github.com/arduino-libraries/Arduino_PortentaMachineControl/assets/45899760/afa032e7-a619-480d-9551-b1c1696ef357)

**Comment by caldecy** [comment]
The version is 4.1.3.
![image](https://github.com/arduino-libraries/Arduino_PortentaMachineControl/assets/33807640/35260030-8da5-489c-a1b0-2f5f0f1c6366)

**Comment by leonardocavagnis** [comment]
Sorry, we don't support [platform.io](http://platform.io/). If you can retry with the latest Arduino IDE or cli and report if the error still occurs we'll be able to help you.

**Comment by jonas-hagen** [comment]
I am facing the same issue:

Arduino Mbed OS Portenta Boards: 4.1.5
Arduinio PortentaMachineControl: 1.0.1

Thermocouples read NaN using the Examples code, RTDs work fine, but are unfortunately not an option for my project.

@leonardocavagnis could you reproduce the problem?

**Comment by leonardocavagnis** [comment]
> I am facing the same issue:
> 
> Official Arduino IDE 2.3.2 Arduino Mbed OS Portenta Boards: 4.1.5 Arduinio PortentaMachineControl: 1.0.1
> 
> Thermocouples read NaN using the Examples code, RTDs work fine, but are unfortunately not an option for my project.
> 
> Old Arduino_MachineControl library works, reads correct TC temperatures.
> 
> @leonardocavagnis could you reproduce the problem?

Sorry for the late reply. 
Could you please provide more details about the code you're using, your hardware setup, and anything else that could help me better understand the issue?
Thanks!

**Comment by jonas-hagen** [comment]
Thanks for the reply!
I am using the bare examples code that ships with the respective versions of the PortentaMachineControl and Arduino_MachineControl and exactly the hardware setup described here: https://docs.arduino.cc/tutorials/portenta-machine-control/rtd-thermocouple-pmc/ there is no custom additions or other peripherals connected.

**Comment by SamThomasIaaps** [comment]
Hello,

I am also facing the same issue.

Version 1.2.1 of the library, Arduino IDE 2.3.4, MBED OS Portenta V4.2.1

I have one thermocouple (non-grounded) connected to channel 0 and the base example shows all NaN. I have also repeated this with the same response on two separate boards with the same wiring.

Just uploaded my code using the old library and that is working as usual.

Please let me know if you need anymore info.

Thanks,
Sam

**Comment by leonardocavagnis** [comment]
Hi @SamThomasIaaps and @jonas-hagen,
Thank you for reporting!

I’m currently unable to replicate the problem on my end, so I would need as many details as possible to assist you better. Could you please provide more information about the thermocouple? What is the model? Is it grounded or ungrounded? Is it Type J or Type K?

Which channel are you using? Is your setup aligned with the connections shown below? If possible, please share a picture of your setup.

Thanks

![image](https://github.com/user-attachments/assets/1adcffa0-47da-439a-8c4c-7591649c27a4)

**Comment by leonardocavagnis** [comment]
@jonas-hagen thanks!

The PMC you're using has this chip near the `TEMP PROBES` label. Right?
![Image](https://github.com/user-attachments/assets/df7f16f8-aeb4-4b09-a29a-16d55ee2c5ae)

**Comment by jonas-hagen** [comment]
Yes, it has this chip as far as I can remember. Let me check on Monday.

**Comment by leonardocavagnis** [comment]
Dear @SamThomasIaaps, @jonas-hagen, @caldecy 
Thanks for patience! This PR should fix the mentioned issue 🎉
Please, give it a spin and let me know

**Comment by SamThomasIaaps** [comment]
Hi @leonardocavagnis,

Have just tried it on my with version 1.0.3 and all seems to be working now. 

Thank you for making the fix!

Sam
